packages/app/src/app/hooks/useWalletCapabilities.ts

```diff
@@ -0,0 +1,29 @@
+import { useEffect, useState } from 'react'
+import { WalletCapabilities } from 'viem'
+import { useEIP5792WalletClient } from './useEIP5792WalletClient'
+
+function useWalletCapabilities({ chainId }: { chainId?: number }) {
+  const { data: walletClient } = useEIP5792WalletClient()
+  const [capabilities, setCapabilities] = useState<{ [chainId: number]: WalletCapabilities }>({})
+  const [loading, setLoading] = useState(true)
+
+  useEffect(() => {
+    if (walletClient) {
+      walletClient
+        .getCapabilities()
```

Comment:
Same suggestion as above.

Reply:
You can do this in wagmi from `2.8.0` (it's building now): https://wagmi.sh/react/api/hooks/useCapabilities

Reply:
That was fast! lol I will update to use the wagmi hook 😊